build.sbt

```diff
@@ -182,7 +182,11 @@ lazy val genericBase = crossProject.in(file("generic"))
   )
   .settings(allSettings: _*)
   .settings(
-    libraryDependencies += "com.chuusai" %%% "shapeless" % shapelessVersion
+    libraryDependencies ++= Seq(
+      "com.chuusai" %%% "shapeless" % shapelessVersion,
+      "org.scala-lang" % "scala-compiler" % scalaVersion.value % "provided",
+      "org.typelevel" %%% "macro-compat" % "1.1.1"
```

Comment:
We get this transitively from Shapeless, right? If we need to include it here, we should probably define a `macroCompatVersion` since we use it twice now.